w3lib/url.py

```diff
@@ -34,10 +34,12 @@ def _quote_byte(error):
 
 _safe_chars = RFC3986_RESERVED + RFC3986_UNRESERVED + EXTRA_SAFE_CHARS + b'%'
 
-def safe_url_string(url, encoding='utf8', path_encoding='utf8'):
-    """Convert the given URL into a legal URL by escaping unsafe characters
-    according to RFC-3986.
-
+def safe_url_string(url, encoding='utf8', path_encoding='utf8', quote_path=True):
+    """When quote_path is True, this will convert the given URL into a 
+    legal URL by escaping unsafe characters according to RFC-3986.
+    If quote_path is False, this method will return url as it is.
```

Comment:
I don't think this is correct - "If quote_path is False, this method will return url as it is." - we still do a lot of processing on the URL even if quote_path is False, because it refers only to path handling. I suggest instead describe how path is handled depending on this parameter, and put it after "path component (unless overriden by path_encoding),".

Reply:
@lopuhin , I have made the changes 👍

Reply:
@lopuhin , can you please review.